ui/tests/e2e/workspace.spec.js

```diff
@@ -0,0 +1,230 @@
+  await expect(page.getByRole('button').filter({ hasText: 'Create' })).toBeVisible();
+
+  if (workspaceCount > 0) {
+    // Verify workspace cards are displayed
+    await expect(page.locator('.MuiGrid-container').first()).toBeVisible();
+  } else {
+    // Verify empty state
+    await expect(page.getByText('No workspace available')).toBeVisible();
+    await expect(
+      page.getByText('Click "Create" to establish your first workspace.'),
+    ).toBeVisible();
+  }
+});
+
+test('Switch between grid and table views', async ({ page }) => {
+  // Verify initial grid view
+  await expect(page.locator('.MuiGrid-container').first()).toBeVisible();
+
+  // Switch to table view using aria-label
+  await page.getByRole('button', { name: 'Switch View' }).click();
+
+  // Verify table structure exists (simpler approach)
+  await expect(page.locator('.MuiTable-root').first()).toBeVisible();
+  await expect(page.locator('.MuiTableHead-root').first()).toBeVisible();
+  await expect(page.locator('.MuiTableBody-root').first()).toBeVisible();
+
+  const expectedHeaders = ['Name', 'Description', 'Owner'];
+  for (const header of expectedHeaders) {
+    await expect(
+      page.locator('[class*="MUIDataTableHeadCell-data"]', { hasText: header }).first()
+    ).toBeVisible();
+  }
+
+  // Switch back to grid view
+  await page.getByRole('button', { name: 'Switch View' }).click();
+  await expect(page.locator('.MuiGrid-container').first()).toBeVisible();
+});
+
+test('Create new workspace', async ({ page, provider }) => {
+  const createWorkspaceRes = page.waitForResponse(
+    (response) =>
+      response.url().includes(`${ENV.MESHERY_SERVER_URL}/api/workspaces`) &&
+      response.request().method() === 'POST',
+  );
+
+  // Click Create button
+  await page.getByRole('button').filter({ hasText: 'Create' }).click();
+
+  // Fill workspace form
+  await page.getByLabel('Name').fill('Test Workspace');
+  await page.getByLabel('Description').fill('Test workspace description');
+
+  // Submit form
+  await page.getByRole('button', { name: 'Save' }).click();
+
+  // Wait for response
+  const response = await createWorkspaceRes;
+  const responseBody = await response.json();
+
+  // Verify response payload
+  expect(responseBody).toMatchObject({
+    name: 'Test Workspace',
+    description: 'Test workspace description',
+  });
+
+  // Verify success notification
+  await expect(page.getByText('Workspace "Test Workspace" created')).toBeVisible();
+
+  // Verify the new workspace is visible in the UI
+  const workspaceCard = page.locator('.MuiGrid-container').filter({ 
+    has: page.getByText('Test Workspace')
+  });
+  await expect(workspaceCard).toBeVisible();
+
+  // Verify workspace details in the card
+  await expect(workspaceCard.getByText('Test Workspace').first()).toBeVisible();
+  await expect(workspaceCard.getByText('Test workspace description').first()).toBeVisible();
+});
+
+test('Assign environments to workspace', async ({ page }) => {
```

Comment:
Should we not check for empty state for environments, or designs in transfer lists?

Reply:
Thanks for calling out Sudhanshu, yes I've already made those changes, will push them soon